### PR TITLE
Merge dev to main

### PR DESCRIPTION
**Pull request by wdesmedt**
Extensive amount of changes happened in the dev branch. This PR finally merges them to the main branch so it can be referenced in a regular way by the tutorial on learn.srlinux.dev.

### CODE REVIEW

**Comment by hellt** [comment]
Feel free to merge it in

File: topo_with_sros.yml [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+    - endpoints: ["s2:e1-57", "d1:eth2"]
+    - endpoints: ["s2:e1-121", "d2:eth2"]
+    - endpoints: ["l1:e1-49", "s1:e1-1"]
+    - endpoints: ["l1:e1-48", "s2:e1-1"]
+    - endpoints: ["l2:e1-49", "s1:e1-2"]
+    - endpoints: ["l2:e1-48", "s2:e1-2"]
+    - endpoints: ["l3:e1-49", "s1:e1-3"]
+    - endpoints: ["l3:e1-48", "s2:e1-3"]
+    - endpoints: ["l4:e1-49", "s1:e1-4"]
+    - endpoints: ["l4:e1-48", "s2:e1-4"]
```
Comment (freddebacker) [inline topo_with_sros.yml:166]: Same comment about `ethernet-1/48` as for `topo.yml`

**Review by freddebacker — COMMENTED** [review]
Hi @wdesmedt,
👏 good work!
During review, I focused on diffs with [`27873b88b1b56af98ef78aecb361a8aa1cf5483a`](https://github.com/srl-labs/intent-based-ansible-lab/tree/27873b88b1b56af98ef78aecb361a8aa1cf5483a), not diffs with the current `main`.

File: topo.yml [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+        - ip route add 10.0.0.0/8 via 10.0.3.254
+  links:
+    - endpoints: ["l1:e1-49", "s1:e1-1"]
+    - endpoints: ["l1:e1-48", "s2:e1-1"]
+    - endpoints: ["l2:e1-49", "s1:e1-2"]
+    - endpoints: ["l2:e1-48", "s2:e1-2"]
+    - endpoints: ["l3:e1-49", "s1:e1-3"]
+    - endpoints: ["l3:e1-48", "s2:e1-3"]
+    - endpoints: ["l4:e1-49", "s1:e1-4"]
+    - endpoints: ["l4:e1-48", "s2:e1-4"]
```
[the remark this answers is not in the document]
Comment (wdesmedt) [reply topo.yml:146]: yes, we should. Fix in latest commits

File: topo.yml [hunk 3]
```diff
@@ -0,0 +1,160 @@
+name: 4l2s
+
+mgmt:
+    network: intent-lab
+    ipv4-subnet: 172.20.21.0/24
+
+topology:
+  kinds:
+    srl:
+      image: ghcr.io/nokia/srlinux:24.3.1
```
[the remark this answers is not in the document]
Comment (wdesmedt) [reply topo.yml:10]: Yes, `24.3.1` was pulled from the repo due to some build error. Changed to `24.3.2`

File: README.md [hunk 4]
```diff
@@ -1,3 +1,48 @@
...
+* Limited support for SROS to support DCGW interconnection with the fabric. Use of `ansible.netcommon.netconf` to push/get configuration. See var `sros` in `group_infra` under `./intent/dcgw_test` 
+* bug fixing
+
+[^1]: Intents could have been placed in Ansible `host_vars` and `group_vars` but issues arise when variables are redefined as is the case in host- and group-level intents, due to the hierarchical nature of the variables/device model (e.g. `.network-instance.protocols.bgp` has host-level and group-level definitions but it's a single variable `network-instance`). It requires that variables are *merged* rather than *replaced* which is the default behavior with ansible's `host_vars` and `group_vars`. This behavior can be controlled via `hash_bahaviour=merge` in the `ansible.cfg` file. Ansible development discorages setting this playbook-wide parameter as existing modules and roles assume the default `replace` behavior and may deprecate this option in later releases. To achieve the desired behavior, the `combine` filter is proposed, which is exactly what we're doing in the roles of this playbook.
```
[the remark this answers is not in the document]
Comment (wdesmedt) [reply README.md:48]: Yes. Only reference the tutorial on learn.srlinux.dev.

File: inv/ansible-inventory.yml
Comment (freddebacker) [inline inv/ansible-inventory.yml:1]: The linux hosts section doesn't match the actual deployed nodes anymore. Perhaps update? Or remove?

File: tests/ospf_test_2/fabric.yml
Comment (freddebacker) [inline tests/ospf_test_2/fabric.yml:1]: Wrong spine port-range and cabling endpoints for D3L
- line 47
- lines 51..54

**Review by freddebacker — COMMENTED** [review]
Made some more review comments...

File: intent_examples/infra/underlay_with_level1_intents/group_infra.yml [hunk 5]
```diff
@@ -0,0 +1,214 @@
...
+    system0:
+      admin_state: enable
+  bfd:
+    interfaces:
+      system0.0:
+        admin_state: enable
+        desired_min_tx_interval: 100000
+        detection_multiplier: 3
+        required_min_rx_interval: 100000
+      ethernet-1/{48..49}.0:
```
[the remark this answers is not in the document]
Comment (wdesmedt) [reply intent_examples/infra/underlay_with_level1_intents/group_infra.yml]: fixed

File: intent_examples/infra/underlay_with_level1_intents/group_infra.yml [hunk 6]
```diff
@@ -0,0 +1,214 @@
...
+        admin_state: enable
+        desired_min_tx_interval: 100000
+        detection_multiplier: 3
+        required_min_rx_interval: 100000
+  network_instance:
+    default:
+      interfaces:
+        - system0.0
+        - ethernet-1/{48..49}.0
```
[the remark this answers is not in the document]
Comment (wdesmedt) [reply intent_examples/infra/underlay_with_level1_intents/group_infra.yml]: fixed

File: intent_examples/infra/underlay_with_level1_intents/host_infra.yml [hunk 7]
```diff
@@ -0,0 +1,138 @@
+clab-4l2s-l1:
+  TAGS:
+    - RACK1
+  subinterface:
+    ethernet-1/48.0:
```
[the remark this answers is not in the document]
Comment (wdesmedt) [reply intent_examples/infra/underlay_with_level1_intents/host_infra.yml]: fixed

File: topo_with_sros.yml (hunk 1 above)
Comment (wdesmedt) [reply topo_with_sros.yml:166]: fixed

File: inv/ansible-inventory.yml
Comment (wdesmedt) [reply inv/ansible-inventory.yml:1]: removed linux hosts.

File: intent_examples/infra/underlay_with_fabric_intent/fabric.yml [hunk 8]
```diff
@@ -0,0 +1,87 @@
...
+    location: 'spine'
+    # List of neighbor IPs for Route Reflectors in case of 'external'
+    # neighbor_list:
+    #   - 1.1.1.1
+    #   - 2.2.2.2
+
+  # Spine specific configuration, defines which ports can be used for ISLs
+  spine:
+    clab-4l2s-s[1-2]:
+      isl-ports: ethernet-1/[1-128]
```
[the remark this answers is not in the document]
Comment (wdesmedt) [reply intent_examples/infra/underlay_with_fabric_intent/fabric.yml:41]: fixed

File: tests/dcgw_test/fabric.yml [hunk 9]
```diff
@@ -0,0 +1,99 @@
...
+    location: 'spine'
+    # List of neighbor IPs for Route Reflectors in case of 'external'
+    # neighbor_list:
+    #   - 1.1.1.1
+    #   - 2.2.2.2
+
+  # Spine specific configuration, defines which ports can be used for ISLs
+  spine:
+    clab-4l2s-s[1-2]:
+      isl-ports: ethernet-1/[1-128]
```
[the remark this answers is not in the document]
Comment (wdesmedt) [reply tests/dcgw_test/fabric.yml:41]: fixed

File: tests/dcgw_test/fabric.yml [hunk 10]
```diff
@@ -0,0 +1,99 @@
...
+
+  # Physical cabling layout between network devices
+  fabric_cabling:
+    - endpoints: ["clab-4l2s-s1:e1-57", "clab-4l2s-d1:1/1/c3/1"]
+    - endpoints: ["clab-4l2s-s1:e1-121", "clab-4l2s-d2:1/1/c3/1"]
+    - endpoints: ["clab-4l2s-s2:e1-57", "clab-4l2s-d1:1/1/c2/1"]
+    - endpoints: ["clab-4l2s-s2:e1-121", "clab-4l2s-d2:1/1/c2/1"]
```
[the remark this answers is not in the document]
Comment (wdesmedt) [reply tests/dcgw_test/fabric.yml:48]: fixed

File: topo_with_sros.yml [hunk 11]
```diff
@@ -0,0 +1,180 @@
+name: 4l2s
+
+mgmt:
+    network: intent-lab
+    ipv4-subnet: 172.20.21.0/24
+
+topology:
+  kinds:
+    srl:
+      image: ghcr.io/nokia/srlinux:24.3.1
```
[the remark this answers is not in the document]
Comment (wdesmedt) [reply topo_with_sros.yml:10]: changed to `24.3.2`

File: topo_with_sros.yml [hunk 12]
```diff
@@ -0,0 +1,180 @@
...
+        - ip link set eth2 master bond0
+        - ip link set eth1 up
+        - ip link set eth2 up
+        - ip link set bond0 up
+        - ip route add 10.0.0.0/8 via 10.0.3.254
+  links:
+    - endpoints: ["s1:e1-57", "d1:eth3"]
+    - endpoints: ["s1:e1-121", "d2:eth3"]
+    - endpoints: ["s2:e1-57", "d1:eth2"]
+    - endpoints: ["s2:e1-121", "d2:eth2"]
```
[the remark this answers is not in the document]
Comment (wdesmedt) [reply topo_with_sros.yml:158]: fixed
```
    - endpoints: ["s1:e1-32", "d1:eth3"]
    - endpoints: ["s1:e1-31", "d2:eth3"]
    - endpoints: ["s2:e1-32", "d1:eth2"]
    - endpoints: ["s2:e1-31", "d2:eth2"]
```

**Pull request merged**